### PR TITLE
Change the screen reader semantics of the Likert component

### LINKED ISSUE #1037: WAVE reporting errors on Likert component
### Description of the bug

NOTE: The accessibility for screen reader (VoiceOver) is working as intended using aria-labels. This is an issue noticed by the accessibility testing tool WAVE.

The testing tool Wave is giving a warning that the component contain labels, but that they do not have any content. It has been tested with a screen reader and every bit of information in relation to the radio button was read. (Tested with VoiceOver) Maybe look into if the RadioButtons should be structured in another way?

image
image

### Steps To Reproduce

1. Go to an app with a likert component
2. Run wave or another testing tool

### Additional Information

_No response_

### PR DESCRIPTION
**Pull request by mikaelrss**
## Description
This PR fixes an issue where the label of each individual radio button in the `Likert` was empty. This PR addresses that issue, and also does some other fixes.

**Changes:**
- Add text label to radio buttons which are visible for screen readers only
- Refactor the code such that the `<table>` element which encapsulates the `Likert`, does not behave like a table for screen readers. The TL;DR for why I did this, is that the component is only visually represented as a table, but is in fact just a collection of radio groups. 
image This case, for instance will be read by NVDA like this with our current implementation: "Table. Gjør du leksene dine? Nesten Alltid. Gjør du leksene dine? Nesten Alltid Radio button checked 1 of 1", where as changing the implementation such that screen readers ignore the table elements, and designating each table row the `role="radiogroup"`, NVDA reads it like this: "Gjør du leksene dine? Grouping. Nesten Alltid. Radio button checked 2 of 5". This also means that the screen reader behaves the same whether the page is displayed in mobile or desktop view. More information about the reasoning behind this can be fond [here](https://www.washington.edu/accesstech/2021/09/28/online-survey-tools/#:~:text=The%20Likert%20question,this%20interface%20correctly.)
- Changed the styling of the table such that each table header (except for the inital one) takes exactly 100px width. This ensures that the `Likert` options are aligned underneath each other no matter how many `Likert` components a page has. I also centered the text of the header and the radio buttons. (This was altered in `v3.68.2` and introduced by [this change](https://github.com/Altinn/app-frontend-react/pull/1536/files#diff-bccde60356bf20046e273ef206ae1ffb7271d25a29f01278c4ee8839103f0d38L25)). Udir have already implemented this fixed width change in their forms by overwriting our CSS, so changing this should be safe. 

## Related Issue(s)

- closes #1037 

## Verification/QA

- Manual functionality testing
  - [x] I have tested these changes manually
  - [ ] Creator of the original issue (or service owner) has been contacted for manual testing (or will be contacted when released in alpha)
  - [ ] No testing done/necessary
- Automated tests
  - [ ] Unit test(s) have been added/updated
  - [ ] Cypress E2E test(s) have been added/updated
  - [ ] No automatic tests are needed here (no functional changes/additions)
  - [ ] I want someone to help me make some tests
- UU/WCAG ([follow these guidelines](https://aksel.nav.no/god-praksis/artikler/utvikling?tema=universell-utforming) until we have our own)
  - [x] I have tested with a screen reader/keyboard navigation/automated wcag validator
  - [ ] No testing done/necessary (no DOM/visual changes)
  - [ ] I want someone to help me perform accessibility testing
- User documentation @ [altinn-studio-docs](https://github.com/Altinn/altinn-studio-docs)
  - [ ] Has been added/updated
  
  - [x] No functionality has been changed/added, so no documentation is needed
  - [ ] I will do that later/have created an issue
  
- Support in Altinn Studio
  - [ ] Issue(s) created for support in Studio
  
  - [x] This change/feature does not require any changes to Altinn Studio
- Sprint board
  - [x] The original issue (or this PR itself) has been added to the Team Apps project and to the current sprint board
  - [ ] I don't have permissions to do that, please help me out
- Labels
  - [x] I have added a `kind/*` label to this PR for proper release notes grouping
  - [ ] I don't have permissions to add labels, please help me out

### CODE REVIEW

File: src/layout/Likert/LikertComponent.module.css [hunk 1]
```diff
@@ -16,13 +21,22 @@
   width: 100%;
 }
 
+.likertRadioButton {
+  display: flex;
+  align-items: center;
+  justify-content: center;
+}
+
+.likertRadioButton label {
```
Comment (mikaelrss) [inline src/layout/Likert/LikertComponent.module.css:30]: The `label` has a gap that comes from the design system. This made the radio button ever so slightly not centered in the table cell.

File: src/layout/Likert/RepeatingGroupsLikertContainer.tsx [hunk 2]
```diff
@@ -108,13 +108,14 @@ export const RepeatingGroupsLikertContainer = ({ node }: RepeatingGroupsLikertCo
...
             className={classes.likertTable}
+            role='group'
           >
             <Table.Head
               id={`likert-table-header-${id}`}
               className={classes.likertTableHeader}
             >
               <Table.Row>
-                <Table.HeaderCell>
+                <Table.HeaderCell aria-hidden='true'>
```
Comment (mikaelrss) [inline src/layout/Likert/RepeatingGroupsLikertContainer.tsx:118]: Hiding the header cells from screen readers, as they are confusing to the screen reader user. They are merely a visual aide for visual users

File: test/e2e/integration/frontend-test/mobile.ts [hunk 3]
```diff
@@ -106,7 +106,7 @@ function ensureTableHasNumColumns(tableContainer: string, numColumns: number) {
 }
 
 function testLikert() {
-  likertPage.selectRequiredRadiosInMobile();
+  likertPage.selectRequiredRadios();
```
Comment (mikaelrss) [inline test/e2e/integration/frontend-test/mobile.ts:109]: Mobile and desktop behave the same 😄

File: src/layout/Likert/RepeatingGroupsLikertContainerTestUtils.tsx [hunk 4]
```diff
@@ -254,28 +254,23 @@ export function ContainerTester(props: { id: string }) {
...
+  screen.getByRole('group');
 
   for (const option of defaultMockOptions) {
-    const columnHeader = screen.getByRole('columnheader', {
-      name: new RegExp(option.label),
-    });
-    expect(columnHeader).toBeInTheDocument();
+    const allElements = screen.getAllByRole('radio', { name: option.label });
+    for (const element of allElements) {
+      expect(element).toBeInTheDocument();
```
Comment (mikaelrss) [inline src/layout/Likert/RepeatingGroupsLikertContainerTestUtils.tsx:262]: Each radio button will place an element with the word "Bra" in the dom. We don't have any role to distinguish this text on any more.

**Review by bjosttveit — APPROVED** [review]
Great stuff!

**Pull request merged**